src/Microsoft.AspNetCore.Sockets.Client.Http/HttpOptions.cs

```diff
@@ -0,0 +1,16 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Net.Http;
+
+namespace Microsoft.AspNetCore.Sockets.Client.Http
+{
+    public class HttpOptions
+    {
+        public HttpMessageHandler HttpMessageHandler { get; set; }
+        public IReadOnlyCollection<KeyValuePair<string, string>> Headers { get; set; }
+        public Func<string> JwtBearerTokenFactory { get; set; }
```

Comment:
This is too specific

Reply:
In what sense it is too specific?

Reply:
I'm not sold on having anything beyond headers on this API.

Reply:
I started with that. However JWT token may change while the connection is open so the user will have to update the headers collection. How can they do that in a thread safe manner? Do you want to expose headers as ConcurrentDictionary?

Reply:
Also, if this is treated like all other headers it will be on the user to format it correctly...